docs/developer-docs/latest/developer-resources/database-apis-reference/graphql-api.md

```diff
@@ -0,0 +1,471 @@
+---
+title: GraphQL API - Strapi Developer Documentation
+description: Use a GraphQL endpoint in your Strapi project to fetch and mutate your content.
+sidebarDepth: 3
+---
+<!-- TODO: update SEO -->
+
+# GraphQL API
+
+::: strapi GraphQL plugin required
+To use the GraphQL API, install the [GraphQL](/developer-docs/latest/development/plugins/graphql.md) plugin.
+:::
+
+## Unified response format
+
+Whatever the query, the response can contain the following fields:
+
+- `data`: the response data itself, which could be:
+  - a single entry, as an object with the following keys:
+    - `id` (number)
+    - `attributes` (object)
+    - `meta` (object)
+  - a list of entries, as an array of objects
+  - a custom response
+
+- `meta`(object): information about pagination, publication state, available locales…
+<!-- TODO: create an entry in the docs to list all errors -->
+
+## Queries
+
+We assume that the [Shadow CRUD](/developer-docs/latest/development/plugins/graphql.md#shadow-crud) feature is enabled. For each model, the GraphQL plugin auto-generates queries and mutations which just fit to your needs.
+
+### Fetch a single entry
+
+Single entries can be found by their `id`.
+
+:::request Example query: Find the entry with id 1, only the French fields
+```graphql
+query {
+  document(id: 1, locale: "fr", publicationState: LIVE) {
+    data {
+      id
+      attributes {
+        title
+        locale
+        categories {
+          data {
+            id
+            attributes {
+              name
+            }
+          }
+          meta {
+            pagination {
+              page
+              pageSize
+              total
+              pageCount
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+:::
+
+### Fetch multiple entries
+
+:::request Example query: Find all documents and populate 'categories' relation with the `name` attribute
+
+```graphql
+query {
+  documents {
+    data {
+      id
+      attributes {
+        title
+        locale
+        categories {
+          data {
+            id
+            attributes {
+                name
+            }
+          }
+        }
+      }
+    }
+    meta {
+      pagination {
+        page
+        pageSize
+        total
+        pageCount
+      }
+    }
+  }
+}
+
+```
+
+:::
+
+<!-- ? is this part 👇 still relevant? -->
```

Comment:
Eventually yes, but since we haven't merged the GraphQL PR I'm not sure how it will function yet.

CC @Convly

Reply:
This part is still relevant and works the same as before.
The updated syntax looks like the following:

```graphql
query {
  restaurants {
    data {
      attributes {
        dz {
          __typename
          ...on ComponentDefaultClosingperiod {
            label
          }
        }
      }
    }
  }
}
```